pkgs/development/python-modules/nianet/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, matplotlib
+, niapy
+, numpy
+, poetry-core
+, pytestCheckHook
+, pythonOlder
+, scikit-learn
+, torch
+}:
+
+buildPythonPackage rec {
+  pname = "nianet";
+  version = "1.1.1";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "SasoPavlic";
+    repo = pname;
+    rev = "Latest";
+    sha256 = "sha256-df3kOWs17fT74dyKF2ZqPNOhMeH8yDEDkJOsMtFNHsM=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    niapy
+    numpy
+    scikit-learn
+    torch
+  ];
+
+  # no serious tests | some tests are incorrect | skip for now
+  # reported to the upstream
+  #checkInputs = [
```

Comment:
Those seem to be fine but small https://github.com/SasoPavlic/NiaNet/blob/main/tests/test_nianet.py

Reply:
Nianet upgraded to the latest release. Tests are now enabled.

Reply:
@SuperSandro2000, thanks.